router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/marketobulkupload.go

```diff
@@ -0,0 +1,325 @@
+)
+
+type MarketoBulkUploader struct {
+	destName          string
+	destinationConfig map[string]interface{}
+	TransformUrl      string
+	PollUrl           string
+	logger            logger.Logger
+}
+
+func NewManager(destination *backendconfig.DestinationT) (*MarketoBulkUploader, error) {
+	marketoBulkUpload := &MarketoBulkUploader{destName: "MARKETO_BULK_UPLOAD", destinationConfig: destination.DestinationDefinition.Config, PollUrl: "/pollStatus", TransformUrl: config.GetString("DEST_TRANSFORM_URL", "http://localhost:9090")}
+	return marketoBulkUpload, nil
+}
+
+var pkgLogger logger.Logger
+
+func init() {
+	pkgLogger = logger.NewLogger().Child("asyncdestinationmanager").Child("marketobulkupload")
+}
+
+func (b *MarketoBulkUploader) Poll(pollInput common.AsyncPoll) common.PollStatusResponse {
+	payload, err := json.Marshal(pollInput)
+	if err != nil {
+		// needs to be retried
+		// transformerConnectionStatus := 500
+		b.logger.Error("Error in Marshalling Poll Input: %w", err)
+		return common.PollStatusResponse{
+			StatusCode: 500,
+			HasFailed:  true,
+		}
+	}
+	bodyBytes, transformerConnectionStatus := misc.HTTPCallWithRetryWithTimeout(b.TransformUrl+b.PollUrl, payload, config.GetDuration("HttpClient.marketoBulkUpload.timeout", 30, time.Second))
+	if transformerConnectionStatus != 200 {
+		return common.PollStatusResponse{
+			StatusCode: transformerConnectionStatus,
+			HasFailed:  true,
+		}
+	}
+	var asyncResponse common.PollStatusResponse
+	err = json.Unmarshal(bodyBytes, &asyncResponse)
+	if err != nil {
+		// needs to be retried
+		// transformerConnectionStatus := 500
+		b.logger.Error("Error in Unmarshalling Poll Response: %w", err)
+		return common.PollStatusResponse{
+			StatusCode: 500,
+			HasFailed:  true,
+		}
+	}
+	return asyncResponse
+}
+
+func GenerateFailedPayload(destConfig map[string]interface{}, jobs []*jobsdb.JobT, importID, destType, csvHeaders string) []byte {
+	var failedPayloadT common.AsyncFailedPayload
+	failedPayloadT.Input = make([]map[string]interface{}, len(jobs))
+	failedPayloadT.Config = destConfig
+	for index, job := range jobs {
+		failedPayloadT.Input[index] = make(map[string]interface{})
+		var message map[string]interface{}
+		metadata := make(map[string]interface{})
+		err := json.Unmarshal([]byte(common.GetTransformedData(job.EventPayload)), &message)
+		if err != nil {
+			return nil
+		}
+		metadata["job_id"] = job.JobID
+		failedPayloadT.Input[index]["message"] = message
+		failedPayloadT.Input[index]["metadata"] = metadata
+	}
+	failedPayloadT.DestType = strings.ToLower(destType)
+	failedPayloadT.ImportId = importID
+	failedPayloadT.MetaData = common.MetaDataT{CSVHeaders: csvHeaders}
+	payload, err := json.Marshal(failedPayloadT)
+	if err != nil {
+		return nil
+	}
+	return payload
+}
+
+func (b *MarketoBulkUploader) GetUploadStats(UploadStatsInput common.FetchUploadJobStatus) common.GetUploadStatsResponse {
+	transformUrl := config.GetString("DEST_TRANSFORM_URL", "http://localhost:9090")
```

Comment:
`b` already has transformUrl. Can we use that?

Reply:
updated